generator/src/main/java/com/algorand/sdkutils/listeners/JavaGenerator.java

```diff
@@ -1,4 +1,933 @@
+
+            Tools.addImport(imports, "java.util.Objects"); // used by Objects.deepEquals
+
+            Tools.addImport(imports, "com.algorand.algosdk.v2.client.common.PathResponse");
+            Tools.addImport(imports, "com.fasterxml.jackson.annotation.JsonProperty");
+
+            currentModelBuffer = new StringBuilder();
+            if (sDef.doc != null) {
+                JavaGenerator.append(currentModelBuffer, Tools.formatComment(sDef.doc, "", true));
+            }
+            JavaGenerator.append(currentModelBuffer, "public class " + className + " extends PathResponse {\n\n");
+        }
+
+        pendingOpenFile = true;
+        modelPropertyAdded = false;
+    }
+
+    // Returns an iterator in sorted order of the properties (json nodes). 
+    Iterator<Entry<String, TypeDef>> getSortedProperties() {
+        Iterator<Entry<String, TypeDef>>sortedProps = properties.entrySet().iterator();
+        return sortedProps;
+    }
+
+    // Write the properties of the Model class.
+    void writeProperties(StringBuilder buffer) {
+        Iterator<Entry<String, TypeDef>> sorted = getSortedProperties();
+        while (sorted.hasNext()) {
+            Entry<String, TypeDef> prop = sorted.next();
+            TypeDef typeObj = prop.getValue();
+
+            JavaGenerator.setImportOf(typeObj, imports);
+
+            String desc = null;
+            if (typeObj.doc != null) {
+                desc = typeObj.doc;
+                desc = Tools.formatComment(desc, TAB + "", true);
+            }
+
+            // public type
+            if (desc != null) buffer.append(desc);
+            getPropertyWithJsonSetter(typeObj, buffer);
+            buffer.append("\n");
+        }
+    }
+
+    static void writeExpandedDefinitions(TypeDef typeObj, StringBuilder buffer,
+            TreeMap<String, Set<String>> imports, boolean forModel){
+        JavaGenerator.setImportOf(typeObj, imports, forModel);
+        if (typeObj.javaTypeName.equalsIgnoreCase("Address")) {
+            getAddress(typeObj, buffer, forModel);
+        }
+
+        // Enum is not defined here. It is defined separately in a 
+        // different file.
+
+        if (typeObj.rawTypeName.equals("binary") || 
+                typeObj.rawTypeName.equals("byte")) {
+            if (typeObj.javaTypeName.equals("byte[]")) {
+                getBase64Encoded(typeObj, buffer);    
+                return;
+            }
+            getBase64EncodedArray(typeObj, buffer, forModel);
+
+        }
+
+    }
+
+    // getPropertyWithJsonSetter formats the property into java declaration type with 
+    // the appropriate json annotation.
+    void getPropertyWithJsonSetter(TypeDef typeObj, StringBuilder buffer) {
+        String jprop = typeObj.propertyName;
+        String javaName = Tools.getCamelCase(jprop, false);
+        if (typeObj.isOfType("getterSetter")) {
+            writeExpandedDefinitions(typeObj, buffer, imports, true);
```

Comment:
I still don't understand `getterSetter`. Maybe that should be removed from the parser and handled in the switch below. There could be a case for `Address`, `binary`, `byte`, etc, and each SDK can handle them however it see's fit

Reply:
Simple: it means it has a getter/setter method. 
I agree it should not be here. It can be in the java code. Something for the future.